rclcpp_components/include/rclcpp_components/component_manager.hpp

```diff
@@ -233,6 +233,8 @@ class ComponentManager : public rclcpp::Node
 
 private:
   std::weak_ptr<rclcpp::Executor> executor_;
+  std::map<uint64_t, std::shared_ptr<rclcpp::Executor>> executors_;
```

Comment:
i agree with you that 'dedicated_executors_'  is a better name, and we can't use `std::unique_ptr` for it , because it's also used in `executor_threads_`
https://github.com/ros2/rclcpp/blob/e626f978d7e838835d13663e8cf4534a7753ee3c/rclcpp_components/src/component_manager.cpp#L238-L241

Reply:
To pass the raw pointer (`executors_[node_id].get()`) into the lambda could work.

such as,
```cpp
        auto exec = std::make_unique<rclcpp::executors::SingleThreadedExecutor>();
        exec->add_node(node);
        executors_[node_id] = std::move(exec);
        executor_threads_[node_id] = std::make_shared<std::thread>([executor=executors_[node_id].get()]() {executor->spin();});
```